oximeter/collector/src/agent.rs

```diff
@@ -324,48 +324,27 @@ impl OximeterAgent {
         }
     }
 
-    /// Forces a collection from all producers.
-    ///
-    /// Returns once all those values have been inserted into ClickHouse,
-    /// or an error occurs trying to perform the collection.
+    /// Enqueue requests to forces collection from all producers.
     ///
     /// NOTE: This collection is best effort, as the name implies. It's possible
-    /// that we lose track of requests internally, in cases where there are
-    /// many concurrent calls. Callers should strive to avoid this, since it
-    /// rarely makes sense to do that.
+    /// that we successfully enqueue requests for some producers and not all, in
+    /// cases where there are many concurrent calls. This method _does not_ wait
+    /// for the requested collections to be performed; it is "fire and forget".
+    /// avoid this, since it rarely makes sense to do that.
```

Comment:
Is there any other use case for this method other than for integration testing? If there isn't, perhaps it'd be good to mention that in the comment as well?

Reply:
I don't know of one, but also I think it'd be fine to use outside of tests if there was. The only way to call this is to have a handle to a running Oximeter, which means it's only _possible_ to call in test (where the test context gives you that handle) or inside Oximeter itself. I'm not sure why it would need to, but Oximeter could expose a "run all your collections now" endpoint that called this method, I guess?

Reply:
Ok, makes sense, thanks!